### DISCUSSION TITLE
Censor_thresh in grid_stat

### DISCUSSION THREAD

**Discussion by CPKalb** [dtcenter/METplus#1389] [Configuration] [answered]
Hi:
I'm having trouble setting the censor_thresh and censor_val in grid_stat.  My netCDF file marks all values that should be 0 as missing, using the following:
float MOSA(time, z0, y0, x0) ;
                MOSA:valid_min = -3.4028e+38f ;
                MOSA:valid_max = 3.4028e+38f ;
                MOSA:_FillValue = -25.5f ;

I'm using the following to try to set the missing values to 0:
censor_thresh = [<0];
censor_val = [0];

However, this doesn't seem to produce any change in my output.  Is there something different that needs to be done to set the censor_thresh and censor_val for grid_stat?  I was originally setting these in METplus, but I also tried specifically putting them in the GridStatConfig file.

Thanks

**Comment by JohnHalleyGotway** [comment, marked as answer]
@CPKalb I'd recommend testing this using plot_data_plane. I find that to be the easiest way to try out configuration tweaks like this.

Logic can get a bit tricky when applying thresholds to missing data values. When data is read into MET, all missing data values are stored as -9999 internally. So you'll likely need to use that flag value in your censoring step.

To test, I used a sample data file included with the MET release and ran plot_data_plane 3 times:

(1) Plot the raw data:
```
plot_data_plane data/sample_obs/ST4/ST4.2010010112.24h raw.ps 'name="APCP"; level="A24";'
```

(2) Censor less than 0, as you did:
```
plot_data_plane data/sample_obs/ST4/ST4.2010010112.24h censor_lt0.ps 'name="APCP"; level="A24"; censor_thresh = lt0; censor_val = 0;'
```

(3) Censor equal to bad data:
```
plot_data_plane data/sample_obs/ST4/ST4.2010010112.24h censor_eq-9999.ps 'name="APCP"; level="A24"; censor_thresh = eq-9999; censor_val = 0;'
```

Below is a screenshot of these 3 images generated. Notice the large bar of gray (missing data) values on the left side of the raw and censor_lt0 images. Also notice that that gray area is now white in the image on the right. That means that those bad data values have been changed to zeros.

<img width="1286" alt="Screen Shot 2022-02-01 at 10 25 19 AM" src="https://user-images.githubusercontent.com/21087144/152018843-9b00ff31-8d1b-4207-b973-5cf259c0746c.png">

So please try using:
```
censor_thresh = [ ==-9999 ]; // Note that ==-9999 and eq-9999 are both valid thresholds
censor_val    = [0];
```

If that doesn't produce the desired result, please post a sample data file containing the MOSA variable to this discussion. If you zip it first I think GitHub would allow you to do so. Or point me to its location on one of the NCAR machines. And I can go take a closer look.

**Reply by CPKalb** [reply]
This worked.  Thanks!

**Reply by JohnHalleyGotway** [reply]
Thanks for confirming Tina. I'll go ahead and lock conversation on this discussion.
